CONTRIBUTING.md

```diff
@@ -0,0 +1,172 @@
+
+Sometimes there won’t be any issues with these labels. That’s ok! There is
+likely still something for you to work on. If you want to contribute but you
+don’t know where to start or can't find a suitable issue, you can ask for help by reaching out on one of our
+[channels](#ask-for-help).
+
+If you see an issue that you'd like to work on, please post a comment saying
+that you want to work on it and assign yourself.
+Something like "I want to work on this" is fine.
+
+## Ask for Help
+
+The best way to reach us with a question when contributing is to ask on:
+
+* The original github issue
+* Our mailing list [sst-cs-software-management@redhat.com](mailto:sst-cs-software-management@redhat.coms)
+* On IRC [#dnf](irc:/chat.libera.org/#dnf)
+
+## Pull Request Lifecycle
+
+Once you have a contribution ready, follow these steps:
+- Submit the PR
+  - Set the [labels](https://github.com/rpm-software-management/dnf5/labels) accordingly.
+  - If you need help while working on it you could set
+    [work in progess](https://github.com/rpm-software-management/dnf5/labels/work%20in%20progress)
+    and wait for the PR to be triaged.
+  - If you are ready for a code review by one of the maintaineres set the label
+    [ready for review](https://github.com/rpm-software-management/dnf5/labels/ready%20for%20review).
+- Triaging happens usually on Mondays or Thursdays
+- Review!
+  - Discuss the pr with one of the maintainers.
+  - Some changes might be requested.
+  - Review again.
+- Merge!
+
+## Development Environment Setup
+
+#### Requirements
+
+To install build requirements, run:
+```
+dnf builddep dnf5.spec #[--define '_without_<option> 1 ...]
+```
+
+#### Build
+
+To build DNF5, run:
+
+```
+mkdir build
+cd build
+cmake ..
+# or cmake .. [-DWITH_<OPTION>=<ON|OFF> ...]
+make
+```
+
+#### Test
+
+To run the tests, follow the steps to build the code and then run:
+
+```
+# from the 'build' directory
+CTEST_OUTPUT_ON_FAILURE=1 make test
+```
+
+As an alternative, tests can be executed in a verbose mode:
+
+```
+# from the 'build' directory
+make test ARGS='-V'
+```
+
+#### Install
+
+To install the built package, we recommend building an rpm package.
+
+To build rpms using tito, run:
+```
+tito build --rpm --test
+```
+
+To build rpms from git, run:
+
+```
+export PREFIX=$(rpmspec dnf5.spec -q --srpm --qf '%{name}-%{version}'); git archive --format=tar.gz --prefix=$PREFIX/ HEAD > $PREFIX.tar.gz
+rpmbuild -ba --define "_sourcedir $(pwd)" dnf5.spec #[--with=<option>|--without=<option> ...]
+```
+
+## Pull Request Checklist
+
+When you submit your pull request, or you push new commits to it, our automated
+systems will run some checks on your new code. We require that your pull request
+passes these checks, but we also have more criteria than just that before we can
+accept and merge it. We recommend that you check the following things locally
+before you submit your code:
```

Comment:
```suggestion
When you submit your pull request or push new commits to it, our automated
systems will run some checks on your new code. We require that your pull request
passes these checks. However, note that there are more criteria that your pull request must pass before we can
accept and merge it. We recommend that you ensure the following locally
before you submit your code:
```

Reply:
I would also turn the following two criteria into the bulleted list as follows:

- Code builds.
  Unless you 
  [Test your code locally](#development-environment-setup) unless you need our help or want us to review
  your _work in progress_ changes.
- Code is tested.
  If your pull request includes a new or modified functionality within the library, provide
  matching unit tests in the project's test directory to cover the changes. However, if the changes only affect
  the command-line interface, you can provide related CI tests in the
  [ci-dnf-stack](https://github.com/rpm-software-management/ci-dnf-stack) component.
  If you need our assistance, ask the maintainers for help.
- Code passed sanity checks.

And so on. Does that sound ok?

Reply:
sounds good